src/version.hpp

```diff
@@ -0,0 +1,97 @@
+/*!
+A constexpr way to obtain program version information.
+Copyright © 2019, Team Wisp
+
+Permission to use, copy, modify, and/or distribute this software for any purpose with or without fee is hereby granted, provided that the above copyright notice and this permission notice appear in all copies.
+
+THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+
+#define VERSION_FILE "../wisp.version"
+
+#include <cctype>
+#include <cstring>
+#include <cstdint>
+#include <cstdio>
+#include <stdexcept>
+
+namespace wr
+{
+
+	struct Version
+	{
+		std::uint32_t m_major = 0, m_minor = 0, m_patch = 0;
```

Comment:
I still don't like this, even if our code standard doesn't mention it :P your call though :)